pkg/sql/schemachanger/sctest/comparator.go

```diff
@@ -80,8 +80,9 @@ func CompareLegacyAndDeclarative(t *testing.T, ss StmtLineReader) {
 		// state. This step is to account for the known behavior difference between
 		// the two schema changers.
 		// Only run when not in a transaction (otherwise certain DDL combo can make
-		// sql queries issued during modification break; see commit message).
-		if !isInATransaction(ctx, t, legacyConn) && !syntaxError {
+		// sql queries issued during modification break; see commit message) and
```

Comment:
This "see commit message" comment is not very helpful. Now that this line is being modified, the commit message that is responsible for this line of text is changing.

I also think that most of the comment can be simplified. The part that says "Only run when not in a transaction and `line` is a single DDL statement" is a basic summary in English of what the Go code is doing in the single line below. It would be better for the comment to focus on "why" the code is doing this, rather than most of it being an English description of "what" the code is doing.

Reply:
Thanks for the advice/suggestion; I completely agree! I'll merge this as is and refactor the comments in the PR I'm working on right now, since it touches this part of code anyway.